tools/topology/platform/intel/ehl.m4

```diff
@@ -0,0 +1,38 @@
+#
+# ElkhartLake differentiation for pipelines and components
+#
+
+include(`memory.m4')
+
+dnl Memory capabilities for different buffer types on ElkhartLake
```

Comment:
I'm wondering how we decide when to use 'dnl' and when to use '#'? I thought that maybe it was a convention to use 'dnl' for "function prototypes," but here you use it for a normal textual comment. Nothing important of course, but looks a bit weird to me.

Reply:
(I assume noone actually looks in the generated .conf files? # is used for comments which should be present there, and this one in particular is probably not as useful).

I'd say `dnl` is to be used for anything that is related to m4 macros and # for stuff that is about what is actually output by m4.

Reply:
In this particular case, this comment is for the defines themselves, so I don't think it makes sense to have this comment in the generated code (followed by only newlines, as the defines themselves expand to `' -- the empty string)